nav2_bringup/bringup/launch/nav2_bringup_common.py

```diff
@@ -0,0 +1,99 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os, sys
+from ament_index_python.packages import get_package_share_directory, get_package_prefix
+from launch.substitutions import LaunchConfiguration
+from launch.actions import DeclareLaunchArgument
+
+
+# Get the launch directory
+bringup_dir = get_package_share_directory('nav2_bringup')
+launch_dir = os.path.join(bringup_dir, 'launch')
+
+# Create the launch configuration variables
+namespace = LaunchConfiguration('namespace')
+map_yaml_file = LaunchConfiguration('map')
+use_sim_time = LaunchConfiguration('use_sim_time')
+autostart = LaunchConfiguration('autostart')
+params_file = LaunchConfiguration('params_file')
+use_lifecycle_mgr = LaunchConfiguration('use_lifecycle_mgr')
+use_remappings = LaunchConfiguration('use_remappings')
+bt_xml_file = LaunchConfiguration('bt_xml_file')
+
+# Launch configuration variables specific to simulation
+rviz_config_file = LaunchConfiguration('rviz_config_file')
+use_simulator = LaunchConfiguration('use_simulator')
+simulator = LaunchConfiguration('simulator')
+world = LaunchConfiguration('world')
+
+# TODO(orduno) Remove once `PushNodeRemapping` is resolved
+#              https://github.com/ros2/launch_ros/issues/56
+remappings = [((namespace, '/tf'), '/tf'),
+                ((namespace, '/tf_static'), '/tf_static'),
+                ('/scan', 'scan'),
+                ('/tf', 'tf'),
+                ('/tf_static', 'tf_static'),
+                ('/cmd_vel', 'cmd_vel'),
+                ('/map', 'map'),
+                ('/goal_pose', 'goal_pose')]
+
+# Declare the launch arguments
+declare_autostart_cmd = DeclareLaunchArgument(
+    'autostart', default_value='true',
+    description='Automatically startup the nav2 stack')
+
+declare_namespace_cmd = DeclareLaunchArgument(
+    'namespace',
+    default_value='',
+    description='Top-level namespace')
+
+declare_map_yaml_cmd = DeclareLaunchArgument(
+    'map',
+    description='Full path to map yaml file to load')
+
+declare_use_sim_time_cmd = DeclareLaunchArgument(
+    'use_sim_time',
+    default_value='false',
+    description='Use simulation (Gazebo) clock if true')
+
+declare_bt_xml_cmd = DeclareLaunchArgument(
+    'bt_xml_file',
+    default_value=os.path.join(
+        get_package_prefix('nav2_bt_navigator'),
+        'behavior_trees', 'navigate_w_replanning_and_recovery.xml'),
+    description='Full path to the behavior tree xml file to use')
+
+declare_params_file_cmd = DeclareLaunchArgument(
+    'params_file',
+    default_value=os.path.join(bringup_dir, 'params', 'nav2_params.yaml'),
+    description='Full path to the ROS2 parameters file to use for all launched nodes')
+
+declare_use_remappings_cmd = DeclareLaunchArgument(
+    'use_remappings', default_value='false',
+    description='Arguments to pass to all nodes launched by the file')
+
+declare_simulator_cmd = DeclareLaunchArgument(
+    'simulator',
```

Comment:
`headless` would be a more well understood choice of words, I know this was already there, but with a refactor seems like an appropriate change

Reply:
@SteveMacenski You mean instead of gzserver, "headless?"

Reply:
instead of `simulator` (whatever that means) make the bool wheither the simulation is headless or not